pkg/images/file/file_test.go

```
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"context"
+	"testing"
+)
+
+func TestImageProducer(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	producer := ImageProducer("testdata/images.yaml")
+
+	tt := []struct {
+		key   string
+		value string
+	}{
+		{
+			key:   "knative.dev/reconciler-test/cmd/eventshub",
+			value: "quay.io/myregistry/eventshub",
+		},
+		{
+			key:   "knative.dev/reconciler-test/cmd/eventshub2",
+			value: "quay.io/myregistry/eventshub",
+		},
```

If we want to return the error `image for package %s not found in %+v`, could we add a test here too?

---

Sure

---

Done!